sdk/identity/azure-identity/azure/identity/aio/_credentials/base.py

```diff
@@ -0,0 +1,28 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import abc
+
+
+class AsyncCredentialBase(abc.ABC):
+    @abc.abstractmethod
+    async def close(self):
+        pass
+
+    @abc.abstractmethod
+    async def __aenter__(self):
+        pass
+
+    async def __aexit__(self, *args):
+        await self.close()
+
+    def __enter__(self):
```

Comment:
Why add a method that only throws? I would have expected a normal `AttributeError` if you try to use with with a non-contextmanager instance.

Reply:
My intent was to provide guidance for the "~async~ with" oops, following the lead of [AsyncHttpTransport](https://github.com/Azure/azure-sdk-for-python/blob/e59d4b2f4cd7121eaf4af587e3927e6c91e3ed8d/sdk/core/azure-core/azure/core/pipeline/transport/_base_async.py#L188).

Reply:
Huh. I would ask the same question for that as well. I don't think this is a good idea. If nothing else, this makes the type look like it implements the sync context manager until you actually run the code (e.g. MyPy would think that using `with` is perfectly fine.)

Reply:
#9313